server/http_handler.go

```diff
@@ -842,61 +813,52 @@ func NewRegionFrameRange(region *tikv.KeyLocation) (idxRange *RegionFrameRange,
 	return idxRange, nil
 }
 
-// getFirstIdxIdRange return the first table's index range.
-// 1. [start,end] means index id in [start,end] are needed,while record key is not in.
-// 2. [start,~) means index's id in [start,~) are needed including record key index.
-// 3. (~,~) means only record key index is needed
-func (r *RegionFrameRange) getFirstIdxIDRange() (start, end int64) {
-	start = int64(math.MinInt64)
-	end = int64(math.MaxInt64)
-	if r.first.IsRecord {
-		start = end // need record key only,
-		return
+// getRecordFrame returns the record frame of a table. If the table's records
+// are not covered by this frame range, it returns nil.
+func (r *RegionFrameRange) getRecordFrame(tableID int64, dbName, tableName string) *FrameItem {
+	if tableID == r.first.TableID && r.first.IsRecord {
+		return r.first
+	}
+	if tableID == r.last.TableID && r.last.IsRecord {
+		return r.last
 	}
 
-	start = r.first.IndexID
-	if r.first.TableID != r.last.TableID || r.last.IsRecord {
-		return // [start,~)
+	greaterThanFirst := tableID > r.first.TableID || (tableID == r.first.TableID && !r.first.IsRecord)
```

Comment:
I think use tableID >= r.first.TableID should be ok here.

Reply:
`tableID == r.first.TableID` is already handled at L819, so here only check _greaterThan_ rather than _greaterOrEqual_.